modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1987,6 +1987,20 @@ void GDScriptAnalyzer::resolve_function_body(GDScriptParser::FunctionNode *p_fun
 	}
 	p_function->resolved_body = true;
 
+	if (p_function->is_abstract) {
+		// Abstract functions don't have a body.
+		if (p_function->body->statements.size() > 0) {
+			push_error(R"(Abstract function cannot have a body.)", p_function->body);
+		}
+		return;
+	} else if (!p_function->is_static) {
+		// Non-abstract functions must have a body.
+		if (p_function->body->statements.is_empty()) {
+			push_error(R"(Expected previous function to either have ":" followed by a function body, or "@abstract" for an abstract function.)", p_function->body);
```

Comment:
It's odd to mention "previous function" here. It's resolving the very function that is problematic. It should say `Function "func_name()" ...` and use the function itself as the context instead of its body.

Reply:
Good point, we can just use the function as context.